docusaurus/docs/reference/tunnelers/linux/linux-tunnel-troubleshooting.md

```diff
@@ -23,29 +23,69 @@ journalctl -xeu ziti-edge-tunnel.service
 
 ## Intercepting or hosting not working
 
-You may inspect the loaded identity and router info for a running `ziti-edge-tunnel` by dumping it to stdout or the systemd journal with an IPC command, or you may signal to dump the identities' info to a file.
+You may inspect the loaded identity and router info for a running `ziti-edge-tunnel` by dumping it to stdout or the systemd journal with an IPC command. Alternatively, you may signal to dump the identities' info into a file.
+
+```bash
+# dump identities info to std our journal if systemd unit with IPC command
+./ziti-edge-tunnel dump
+```
+
+```bash
+# dump identities info to a file and inspect
+sudo pkill -USR1 -f ziti-edge-tunnel
+less /tmp/ziti-dump.964315.dump
+```
 
-  ```bash
-  # dump identities info to std our journal if systemd unit with IPC command
-  ./ziti-edge-tunnel dump
-  ```
+Inspect the full command-line of the tunneler process
+
+```bash
+$ pgrep -f ziti-edge-tunnel | xargs ps -fww
+UID          PID    PPID  C STIME TTY      STAT   TIME CMD
+ziti      872509       1  0 10:42 ?        Ssl    1:54 /opt/openziti/bin/ziti-edge-tunnel run --verbose= --dns-ip-range=100.64.0.1/12 --identity-dir=/opt/openziti/etc/identities
+```
+
+Confirm the tunneler's tun interface is UP
+
+```bash
+$ ip link sh up tun0
+19: tun0: <POINTOPOINT,MULTICAST,NOARP,UP,LOWER_UP> mtu 1500 qdisc fq_codel state 
+UNKNOWN mode DEFAULT group default qlen 500
+    link/none 
+```
 
-  ```bash
-  # dump identities info to a file and inspect
-  sudo pkill -USR1 -f ziti-edge-tunnel
-  less /tmp/ziti-dump.964315.dump
-  ```
+Find tunneler's nameserver IP
+
+```bash
+$ resolvectl --interface=tun0 dns
+Link 19 (tun0): 100.64.0.2
+```
+
+Query the Ziti nameserver
+
+```bash
+$ dig +noall +answer ssh.ziti.service.example.com @100.64.0.2
+ my.ziti.service.example.com. 60 IN    A       100.64.0.3
+
+$ ncat 100.64.0.3 22 </dev/null 
+SSH-2.0-OpenSSH_7.4
+```
+
+Find the files opened by the tunneler process
+
+```bash
+pgrep -f 'ziti-edge-tunnel' | xargs sudo lsof -Pnp
+```
 
 ## Process keeps crashing
 
-If the tunneller is crashing then it may be crucial to collect and analyze the core dump file. You may need to enable saving core dumps depending upon your OS configuration.
+You must enable saving core dumps for a segmentation fault (crash) to produce a Corefile.
 
-  You can see how dump files are handled by inspecting this file, which is from Ubuntu 20.10.
+  You can see how dump files are handled by inspecting this file from Ubuntu 20.10.
 
-  ```bash
-  $ cat /proc/sys/kernel/core_pattern
-  |/usr/share/apport/apport %p %s %c %d %P %E
-  ```
+```bash
```

Comment:
we should just point out at some other good blog that educates people on how to produce a core file. ideally one that covers "many linuxes" so we don't have to reinvent that wheel

Reply:
Done